src/hivpy/simulation.py

```diff
@@ -39,44 +43,47 @@ def _update_date(self, date):
         self.output_stats["Date"][self.step] = self.latest_date
 
     def _ratio(self, subpop, pop):
-        if sum(pop) != 0:
-            return sum(subpop)/sum(pop)
+        if len(pop) != 0:
+            return len(subpop)/len(pop)
         else:
             return 0
 
-    def _update_HIV_prevalence(self, pop_data):
+    def _update_HIV_prevalence(self, pop: Population):
         # Update total HIV cases and population
-        over_15_idx = selector(pop_data, age=(operator.gt, 15))
-        HIV_pos_idx = selector(pop_data, HIV_status=(operator.eq, True))
+        over_15_idx = pop.get_sub_pop([(col.AGE, operator.ge, 15)])
+        HIV_pos_idx = pop.get_sub_pop([(col.HIV_STATUS, operator.eq, True)])
         self.output_stats["HIV prevalence (tot)"][self.step] = self._ratio(HIV_pos_idx, over_15_idx)
-        self.output_stats["HIV infections (tot)"][self.step] = sum(HIV_pos_idx)
-        self.output_stats["Population (over 15)"][self.step] = sum(over_15_idx)
+        self.output_stats["HIV infections (tot)"][self.step] = len(HIV_pos_idx)
+        self.output_stats["Population (over 15)"][self.step] = len(over_15_idx)
 
         # Update HIV prevalence by sex
-        men_idx = selector(pop_data, sex=(operator.eq, SexType.Male))
+        men_idx = pop.get_sub_pop([(col.SEX, operator.eq, SexType.Male)])
         self.output_stats["HIV prevalence (male)"][self.step] = (
-            self._ratio(HIV_pos_idx & men_idx, men_idx & over_15_idx))
-        women_idx = selector(pop_data, sex=(operator.eq, SexType.Female))
-        self.output_stats["HIV prevalence (female)"][self.step] = self._ratio(
-            HIV_pos_idx & women_idx, women_idx & over_15_idx)
+            self._ratio(pop.get_sub_pop_intersection(men_idx, over_15_idx),
+                        pop.get_sub_pop_intersection(men_idx, over_15_idx)))
```

Comment:
This ratio is always going to be 1 because we're comparing the length of the same sub-population to itself. Shouldn't we include `HIV_pos_idx` in the first intersection?